pkg/controller/migmigration/migmigration_controller.go

```diff
@@ -105,6 +105,9 @@ func (r *ReconcileMigMigration) Reconcile(request reconcile.Request) (reconcile.
 		return reconcile.Result{}, err // requeue
 	}
 
+	// Begin staging conditions.
```

Comment:
My summary of what's going on here:

- At the beginning of reconcile, we start staging conditions and then run validations, which may stage some Conditions
- Additional Conditions will be staged while running the migrate() function
- Before we exit the reconcile, we apply all staged conditions to the resource

Reply:
:+1: